client/js/utils/utils.js

```diff
@@ -0,0 +1,32 @@
+var _ = {};
+
+_.each = function(collection, callback){
+  if(collection instanceof Array){
+    for(var i = 0; i < collection.length; i++){
+      callback(collection[i], i, collection);
+    }
+  }else{
+    for(var key in collection){
+      callback(collection[key], key, collection);
+    }
+  }
+};
+
+_.map = function(collection, callback){
+  var result = [];
+  _.each(collection, function(item, key, collection){
+    result.push(callback(item, key, collection));
+  });
+  return result;
+};
+
+_.reduce = function(collection, callback, accumulator){
+  var start = accumulator === undefined ? 1 : 0;
```

Comment:
let's make the accumulator mandatory to pass in - calling with reduce without explicitly defining an accumulator is unclear, especially when passing in a collection that is a POJO.  the for in loop does not loop through a POJO's keys in a specific order (it depends on implementation) so the accumulator may be initialized to something unexpected